finality-aleph/src/sync/forest/mod.rs

```diff
@@ -537,11 +582,61 @@ where
     pub fn importable(&self, id: &BlockIdFor<J>) -> bool {
         use VertexHandle::Candidate;
         match self.get(id) {
-            Candidate(vertex) => vertex.vertex.importable(),
+            Candidate(vertex) => {
+                vertex.vertex.importable() || vertex.vertex.parent() == Some(&self.favourite)
+            }
             _ => false,
         }
     }
 
+    fn know_most(&self, id: &BlockIdFor<J>) -> impl Iterator<Item = I> {
```

Comment:
Huh, I thought this was necessary for `flat_map` to work, but apparently not at all. Changed to `HashSet`, since this is what it is underneath – although I don't quite get what you mean by using `Option`, that wouldn't really help if the things we want are sets, right?

Reply:
I don't remember if that was my exact intention, but version with `Option` and `impl` could look this:
```
 fn know_most(&self, id: &BlockIdFor<J>) -> impl Iterator<Item = I> {
        match self.get(id) {
            VertexHandle::Candidate(vertex) => Some(vertex.vertex.know_most().clone()).into_iter().flatten(),
            _ => None.into_iter().flatten(),
        }
    }
```